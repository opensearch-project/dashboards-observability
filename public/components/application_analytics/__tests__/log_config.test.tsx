/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { waitFor } from '@testing-library/react';
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import DSLService from 'public/services/requests/dsl';
import React from 'react';
import { coreStartMock } from '../../../../test/__mocks__/coreMocks';
import { LogConfig } from '../components/config_components/log_config';

describe('Log Config component', () => {
  configure({ adapter: new Adapter() });

  it('renders empty log config', async () => {
    const core = coreStartMock;
    const setQuery = jest.fn();
    const setFilters = jest.fn();
    const setStartTime = jest.fn();
    const setEndTime = jest.fn();
    const setIsFlyoutVisible = jest.fn();
    const setNameWithStorage = jest.fn();
    const setDescriptionWithStorage = jest.fn();
    const setQueryWithStorage = jest.fn();
    const setFiltersWithStorage = jest.fn();
    const setAppConfigs = jest.fn();
    const setStartTimeWithStorage = jest.fn();
    const setEndTimeWithStorage = jest.fn();
    const dslService = ({
      http: jest.fn(),
      fetch: jest.fn(),
      fetchIndices: jest.fn(),
      fetchFields: jest.fn(),
    } as unknown) as DSLService;
    const wrapper = mount(
      <LogConfig
        http={core.http}
        chrome={core.chrome}
        parentBreadcrumb={{ text: 'test', href: 'test#/' }}
        query=""
        setQuery={setQuery}
        filters={[]}
        setFilters={setFilters}
        startTime="now-5m"
        setStartTime={setStartTime}
        endTime="now"
        setEndTime={setEndTime}
        indicesExist={true}
        dslService={dslService}
        setIsFlyoutVisible={setIsFlyoutVisible}
        name=""
        description=""
        setNameWithStorage={setNameWithStorage}
        setDescriptionWithStorage={setDescriptionWithStorage}
        setQueryWithStorage={setQueryWithStorage}
        setFiltersWithStorage={setFiltersWithStorage}
        appConfigs={[]}
        setAppConfigs={setAppConfigs}
        setStartTimeWithStorage={setStartTimeWithStorage}
        setEndTimeWithStorage={setEndTimeWithStorage}
        editMode={false}
      />
    );

    wrapper.update();
    await waitFor(() => {
      expect(
        toJson(wrapper, {
          mode: 'deep',
        })
      ).toMatchSnapshot();
    });
  });

  it('renders with query', async () => {
    const core = coreStartMock;
    const setQuery = jest.fn();
    const setFilters = jest.fn();
    const setStartTime = jest.fn();
    const setEndTime = jest.fn();
    const setIsFlyoutVisible = jest.fn();
    const setNameWithStorage = jest.fn();
    const setDescriptionWithStorage = jest.fn();
    const setQueryWithStorage = jest.fn();
    const setFiltersWithStorage = jest.fn();
    const setAppConfigs = jest.fn();
    const setStartTimeWithStorage = jest.fn();
    const setEndTimeWithStorage = jest.fn();
    const dslService = ({
      http: jest.fn(),
      fetch: jest.fn(),
      fetchIndices: jest.fn(),
      fetchFields: jest.fn(),
    } as unknown) as DSLService;
    const wrapper = mount(
      <LogConfig
        http={core.http}
        chrome={core.chrome}
        parentBreadcrumb={{ text: 'test', href: 'test#/' }}
        query="source = openserach_dashboard_sample_logs"
        setQuery={setQuery}
        filters={[]}
        setFilters={setFilters}
        startTime="now-5m"
        setStartTime={setStartTime}
        endTime="now"
        setEndTime={setEndTime}
        indicesExist={true}
        dslService={dslService}
        setIsFlyoutVisible={setIsFlyoutVisible}
        name=""
        description=""
        setNameWithStorage={setNameWithStorage}
        setDescriptionWithStorage={setDescriptionWithStorage}
        setQueryWithStorage={setQueryWithStorage}
        setFiltersWithStorage={setFiltersWithStorage}
        appConfigs={[]}
        setAppConfigs={setAppConfigs}
        setStartTimeWithStorage={setStartTimeWithStorage}
        setEndTimeWithStorage={setEndTimeWithStorage}
        editMode={false}
      />
    );

    wrapper.update();
    await waitFor(() => {
      expect(
        toJson(wrapper, {
          mode: 'deep',
        })
      ).toMatchSnapshot();
    });
  });
});
