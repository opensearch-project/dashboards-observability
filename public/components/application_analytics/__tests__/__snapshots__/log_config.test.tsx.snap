// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Log Config component renders empty log config 1`] = `
<div>
  <div
    className="euiAccordion"
    data-test-subj="logSourceAccordion"
    onToggle={[Function]}
  >
    <div
      className="euiAccordion__triggerWrapper"
    >
      <button
        aria-controls="logSource"
        aria-expanded={false}
        className="euiAccordion__button"
        id="random_html_id"
        onClick={[Function]}
        type="button"
      >
        <span
          className="euiAccordion__iconWrapper"
        >
          <svg
            aria-hidden={true}
            className="euiIcon euiIcon--medium euiIcon-isLoading euiAccordion__icon"
            focusable="false"
            height={16}
            role="img"
            style={null}
            viewBox="0 0 16 16"
            width={16}
            xmlns="http://www.w3.org/2000/svg"
          />
        </span>
        <span
          className="euiIEFlexWrapFix"
        >
          <div
            className="euiText euiText--small"
          >
            <h3>
              Log source
            </h3>
          </div>
          <div
            className="euiSpacer euiSpacer--s"
          />
          <div
            className="euiText euiText--small"
          >
            <div
              className="euiTextColor euiTextColor--subdued"
            >
              Configure your application base query
            </div>
          </div>
        </span>
      </button>
      <div
        className="euiAccordion__optionalAction"
      >
        <button
          className="euiButton euiButton--primary euiButton--small euiButton-isDisabled"
          data-test-subj="clearLogSourceButton"
          disabled={true}
          onClick={[Function]}
          style={
            Object {
              "minWidth": undefined,
            }
          }
          type="button"
        >
          <span
            className="euiButtonContent euiButton__content"
          >
            <span
              className="euiButton__text"
            >
              Clear
            </span>
          </span>
        </button>
      </div>
    </div>
    <div
      aria-labelledby="random_html_id"
      className="euiAccordion__childWrapper"
      id="logSource"
      role="region"
      tabIndex={-1}
    >
      <div>
        <div
          className="euiAccordion__padding--l"
        >
          <div
            className="euiFlexGroup euiFlexGroup--gutterSmall euiFlexGroup--directionColumn euiFlexGroup--responsive"
          >
            <div
              className="euiFlexItem"
            >
              <div
                className="euiCallOut euiCallOut--primary euiCallOut--small"
                id="baseQueryCallout"
              >
                <div
                  className="euiText euiText--extraSmall"
                >
                  <div
                    className="euiTextColor euiTextColor--default"
                  >
                    You can't change the base query after the application is created.
                  </div>
                </div>
              </div>
            </div>
            <div
              className="euiFlexItem"
            >
              <div
                className="euiFormRow"
                id="random_html_id-row"
              >
                <div
                  className="euiFormRow__labelWrapper"
                >
                  <label
                    className="euiFormLabel euiFormRow__label"
                    htmlFor="random_html_id"
                  >
                    Base Query
                  </label>
                </div>
                <div
                  className="euiFormRow__fieldWrapper"
                >
                  <div
                    aria-describedby="random_html_id-help-0"
                    className="euiFlexItem euiFlexItem--flexGrowZero query-area"
                    id="random_html_id"
                    onBlur={[Function]}
                    onFocus={[Function]}
                  >
                    <div
                      aria-expanded={false}
                      aria-haspopup="listbox"
                      aria-labelledby="autocomplete-0-label"
                      className="aa-Autocomplete"
                      id="autocomplete-root"
                      role="combobox"
                    >
                      <textarea
                        aria-autocomplete="both"
                        aria-labelledby="autocomplete-0-label"
                        autoCapitalize="off"
                        autoComplete="off"
                        autoCorrect="off"
                        autoFocus={false}
                        className="euiTextArea euiTextArea--resizeVertical"
                        data-test-subj="searchAutocompleteTextArea"
                        disabled={false}
                        enterKeyHint="search"
                        id="autocomplete-textarea"
                        maxLength={512}
                        onBlur={[Function]}
                        onChange={[Function]}
                        onClick={[Function]}
                        onFocus={[Function]}
                        onKeyDown={[Function]}
                        placeholder="Enter query"
                        rows={6}
                        spellCheck="false"
                        style={
                          Object {
                            "height": "18px",
                            "marginTop": "0px",
                            "padding": "8px",
                          }
                        }
                        type="search"
                        value=""
                      />
                    </div>
                    <button
                      aria-label="pplLinkShowFlyout"
                      className="euiBadge euiBadge-isClickable euiBadge--hollow euiBadge--iconLeft ppl-link ppl-link-light"
                      disabled={false}
                      onClick={[Function]}
                      title="PPL"
                    >
                      <span
                        className="euiBadge__content"
                      >
                        <span
                          className="euiBadge__text"
                        >
                          PPL
                        </span>
                      </span>
                    </button>
                  </div>
                  <div
                    className="euiFormHelpText euiFormRow__text"
                    id="random_html_id-help-0"
                  >
                    The default logs view in the application will be filtered by this query.
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Log Config component renders with query 1`] = `
<div>
  <div
    className="euiAccordion"
    data-test-subj="logSourceAccordion"
    onToggle={[Function]}
  >
    <div
      className="euiAccordion__triggerWrapper"
    >
      <button
        aria-controls="logSource"
        aria-expanded={false}
        className="euiAccordion__button"
        id="random_html_id"
        onClick={[Function]}
        type="button"
      >
        <span
          className="euiAccordion__iconWrapper"
        >
          <svg
            aria-hidden={true}
            className="euiIcon euiIcon--medium euiAccordion__icon"
            focusable="false"
            height={16}
            role="img"
            style={null}
            viewBox="0 0 16 16"
            width={16}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m5.157 13.069 4.611-4.685a.546.546 0 0 0 0-.768L5.158 2.93a.552.552 0 0 1 0-.771.53.53 0 0 1 .759 0l4.61 4.684c.631.641.63 1.672 0 2.312l-4.61 4.684a.53.53 0 0 1-.76 0 .552.552 0 0 1 0-.771Z"
              fillRule="nonzero"
            />
          </svg>
        </span>
        <span
          className="euiIEFlexWrapFix"
        >
          <div
            className="euiText euiText--small"
          >
            <h3>
              Log source
            </h3>
          </div>
          <div
            className="euiSpacer euiSpacer--s"
          />
          <div
            className="euiText euiText--small"
          >
            <div
              className="euiTextColor euiTextColor--subdued"
            >
              Configure your application base query
            </div>
          </div>
        </span>
      </button>
      <div
        className="euiAccordion__optionalAction"
      >
        <button
          className="euiButton euiButton--primary euiButton--small euiButton-isDisabled"
          data-test-subj="clearLogSourceButton"
          disabled={true}
          onClick={[Function]}
          style={
            Object {
              "minWidth": undefined,
            }
          }
          type="button"
        >
          <span
            className="euiButtonContent euiButton__content"
          >
            <span
              className="euiButton__text"
            >
              Clear
            </span>
          </span>
        </button>
      </div>
    </div>
    <div
      aria-labelledby="random_html_id"
      className="euiAccordion__childWrapper"
      id="logSource"
      role="region"
      tabIndex={-1}
    >
      <div>
        <div
          className="euiAccordion__padding--l"
        >
          <div
            className="euiFlexGroup euiFlexGroup--gutterSmall euiFlexGroup--directionColumn euiFlexGroup--responsive"
          >
            <div
              className="euiFlexItem"
            >
              <div
                className="euiCallOut euiCallOut--primary euiCallOut--small"
                id="baseQueryCallout"
              >
                <div
                  className="euiText euiText--extraSmall"
                >
                  <div
                    className="euiTextColor euiTextColor--default"
                  >
                    You can't change the base query after the application is created.
                  </div>
                </div>
              </div>
            </div>
            <div
              className="euiFlexItem"
            >
              <div
                className="euiFormRow"
                id="random_html_id-row"
              >
                <div
                  className="euiFormRow__labelWrapper"
                >
                  <label
                    className="euiFormLabel euiFormRow__label"
                    htmlFor="random_html_id"
                  >
                    Base Query
                  </label>
                </div>
                <div
                  className="euiFormRow__fieldWrapper"
                >
                  <div
                    aria-describedby="random_html_id-help-0"
                    className="euiFlexItem euiFlexItem--flexGrowZero query-area"
                    id="random_html_id"
                    onBlur={[Function]}
                    onFocus={[Function]}
                  >
                    <div
                      aria-expanded={false}
                      aria-haspopup="listbox"
                      aria-labelledby="autocomplete-1-label"
                      className="aa-Autocomplete"
                      id="autocomplete-root"
                      role="combobox"
                    >
                      <textarea
                        aria-autocomplete="both"
                        aria-labelledby="autocomplete-1-label"
                        autoCapitalize="off"
                        autoComplete="off"
                        autoCorrect="off"
                        autoFocus={false}
                        className="euiTextArea euiTextArea--resizeVertical"
                        data-test-subj="searchAutocompleteTextArea"
                        disabled={false}
                        enterKeyHint="search"
                        id="autocomplete-textarea"
                        maxLength={512}
                        onBlur={[Function]}
                        onChange={[Function]}
                        onClick={[Function]}
                        onFocus={[Function]}
                        onKeyDown={[Function]}
                        placeholder="Enter query"
                        rows={6}
                        spellCheck="false"
                        style={
                          Object {
                            "height": "18px",
                            "marginTop": "0px",
                            "padding": "8px",
                          }
                        }
                        type="search"
                        value="source = openserach_dashboard_sample_logs"
                      />
                    </div>
                    <button
                      aria-label="pplLinkShowFlyout"
                      className="euiBadge euiBadge-isClickable euiBadge--hollow euiBadge--iconLeft ppl-link ppl-link-light"
                      disabled={false}
                      onClick={[Function]}
                      title="PPL"
                    >
                      <span
                        className="euiBadge__content"
                      >
                        <span
                          className="euiBadge__text"
                        >
                          PPL
                        </span>
                      </span>
                    </button>
                  </div>
                  <div
                    className="euiFormHelpText euiFormRow__text"
                    id="random_html_id-help-0"
                  >
                    The default logs view in the application will be filtered by this query.
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
